Logger/Source/Log.swift

```diff
@@ -0,0 +1,81 @@
+import CocoaLumberjackSwift
+
+public struct Log {
+
+    public func verbose(_ message: @autoclosure () -> String, file: StaticString = #file, function: StaticString = #function, line: UInt = #line) {
+        Logger.shared.log(.verbose, message, file: file, function: function, line: line)
+    }
+
+    static public func debug(_ message: @autoclosure () -> String, file: StaticString = #file, function: StaticString = #function, line: UInt = #line) {
+        Logger.shared.log(.debug, message, file: file, function: function, line: line)
+    }
+
+    static public func info(_ message: @autoclosure () -> String, file: StaticString = #file, function: StaticString = #function, line: UInt = #line) {
+        Logger.shared.log(.info, message, file: file, function: function, line: line)
+    }
+
+    static public func warn(_ message: @autoclosure () -> String, file: StaticString = #file, function: StaticString = #function, line: UInt = #line) {
+        Logger.shared.log(.warn, message, file: file, function: function, line: line)
+    }
+
+    static public func error(_ message: @autoclosure () -> String, file: StaticString = #file, function: StaticString = #function, line: UInt = #line) {
+        Logger.shared.log(.error, message, file: file, function: function, line: line)
+    }
+
+}
+
+/// A logger intended to be used as a singleton.
+public struct Logger {
+
+    public static let shared = Logger()
+
+    private let fileLogger: DDFileLogger
+
+    private init() {
+        let logFormatter = LogFormatter()
+
+        fileLogger = DDFileLogger()
+        fileLogger.rollingFrequency = 60 * 60 * 24 // 24 hours
+        fileLogger.logFileManager.maximumNumberOfLogFiles = 7
+        fileLogger.logFormatter = logFormatter
+        DDLog.add(fileLogger, with: .info)
+
+        let ttyLogger = DDOSLogger.sharedInstance!
```

Comment:
This one is tricky.

The implementation of that shared instance can never return nil, it's just lacking nullability annotations in the Objective-C code. CocoaLumberjack hasn't had a new release in a while, but they recently switched maintainers and it looks like they're picking it back up, so hopefully that gets fixed soon. I was thinking about submitting a PR, but it'd be best to add nullability annotations everywhere that it's lacking, and that could be a huge time sink.

With that in mind, I think for now some form of force unwrapping is OK for this version of CocoaLumberjack. It might be more clear if I were to use a `guard` and add a comment, so I'll do that now. I think that's a good call. If CocoaLumberjack is ever updated in this repo we'd have to verify that it's still not able to crash, but that should be well tested since logging should be used on every launch.

Are you aware of an alternative logging library that you could recommend that has better Swift support (perhaps written in Swift)?

Reply:
Actually I should be able to avoid using the `sharedInstance` completely and just create a new instance. I'll do that.